src/nvim/mbyte.c

```diff
@@ -117,6 +118,27 @@ struct interval {
 # include "mbyte.c.generated.h"
 #endif
 
+/// To speed up BYTELEN() we fill a table with the byte lengths whenever
+/// enc_utf8 or enc_dbcs changes.
+static uint8_t mb_bytelen_tab[256];
+
+/// @return byte length of character that starts with byte "b".
+/// Returns 1 for a single-byte character.
+/// mb_byte2len_check() can be used to count a special key as one byte.
+uint8_t mb_byte2len(uint8_t b)
```

Comment:
https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html#Common-Function-Attributes talks only about reading globals, it doesn't say anything about the mutability of the globals. I suspect the "can be called less times than written" may cause problems in this case.